### PR TITLE
Repository cloning implementation

### LINKED ISSUE #17: Cloning a Repository
It should be possible for the app to clone a public repository from GitHub. This can be the first iteration towards our end goal - simultaneously cloning numerous public/private repositories. Proposed features:

- A backend component wrapping around a library such as [nodegit](http://www.nodegit.org/), which accepts a repository URL, path and notifies the caller about the progress and on completion.

- Integration with the existing mockup developed in #15. This includes specifying the repository URL, selecting the default clone destination on the settings page, changing the destination on the confirmation page, and viewing the job status on the progress page.

CC @johndbritton @tarebyte

### PR DESCRIPTION
**Pull request by NickTikhonov**
This PR introduces a module for cloning repositories and aims to resolve #17 

CC @johndbritton @tarebyte

### CODE REVIEW

File: app/lib/__tests__/cloneutils-test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+jest.unmock("../cloneutils.js")
+
+import { clone } from "../cloneutils.js"
+import rmdir from "rimraf"
+import sinon from "sinon"
+
+const fs = require("fs")
+
+const TEST_REPO = "https://github.com/education/classroom-desktop"
+const TEST_FAKE_REPO = "https://github.com/education/a-repo-that-will-never-exist"
```
Comment (tarebyte) [inline app/lib/__tests__/cloneutils-test.js]: We hope 😁

File: app/lib/cloneutils.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+//      "https://github.com/education/classroom-desktop",
+//      "/tmp/desktop-classroom",
+//      (percentage) => {
+//        console.log("Percentage complete: " + percentage + "%")
+//      }
+//    ).then(() => {
+//      console.log("Finished cloning")
+//    })
+//
+// Returns a Promise
```
Comment (tarebyte) [inline app/lib/cloneutils.js]: ❤️  the use of TomDoc here

File: app/lib/__tests__/cloneutils-test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        TEST_FAKE_REPO,
+        DESTINATION,
+        () => {}
+      ).then(() => {
+        fail("Clone should not succeed")
+      }).catch(() => {
+        done()
+      })
+    })
+  })
```
Comment (tarebyte) [inline app/lib/__tests__/cloneutils-test.js]: Can we add a test to for if the teacher tries to overwrite and exiting repo?

**Comment by tarebyte** [comment]
@NickTikhonov this is coming along nicely, I left a few comments.

**Comment by NickTikhonov** [comment]
Thanks @tarebyte - I'll add that test case 👍

File: app/lib/__tests__/cloneutils-test.js (hunk 1 above)
Comment (NickTikhonov) [inline app/lib/__tests__/cloneutils-test.js]: I've never seen anyone define constants in camelCase, mostly uppercase. Seems to be the convention in most languages :)

File: app/lib/__tests__/cloneutils-test.js (hunk 3 above)
Comment (NickTikhonov) [inline app/lib/__tests__/cloneutils-test.js]: Done! seems like `node-git` already throws an error in that case.

**Pull request merged**